src/xdocs/config_filters.xml

```diff
@@ -991,6 +991,216 @@ public class InputTest {
       </subsection>
     </section>
 
+    <section name="SuppressionXpathSingleFilter">
+      <subsection name="Description" id="SuppressionXpathSingleFilter_Description">
+        <p>Since Checkstyle 8.16</p>
+        <p>
+          Filter <code>SuppressionXpathSingleFilter</code> suppresses audit events for
+          Checks violations in the specified file, class, method or variable.
+        </p>
+        <p>
+            Attention: This filter only supports single suppression, and will need
+            multiple instances if users wants to suppress multiple violations.
+        </p>
+      </subsection>
+      <subsection name="Properties" id="SuppressionXpathSingleFilter_Properties">
+        <table>
+            <tr>
+                <th>name</th>
+                <th>description</th>
+                <th>type</th>
+                <th>default value</th>
+                <th>since</th>
+            </tr>
+            <tr>
+                <td>files</td>
+                <td>a Regular Expression matched against the file name associated
+                    with an audit event. It is optional.</td>
+                <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                <td><code>null</code></td>
+                <td>8.16</td>
+            </tr>
+            <tr>
+                <td>checks</td>
+                <td>a Regular Expression matched against the name of the check
+                    associated with an audit event. Optional as long as id or message
+                    is specified.</td>
```

Comment:
Based on current code, all fields are optional. This is because you are bypassing the required checks that are in another class and are not in `XpathFilter`.

@romani Is this ok, or do we want to enforce required fields?

Reply:
@romani ping.

Reply:
> or do we want to enforce required fields?

In checkstyle we do not have required fields. All fields have default values and Check is not failing with exception if property is not defined.
I would remove all "Optional".
Property is always defined by default. If user define its own value, we use users value.

Reply:
@rnveach I removed the whole optional words. Is that OK?

Reply:
done. The "optional" words have all been removed.